### PR TITLE
GitHub Actions

### PR DESCRIPTION
**Pull request by Rayrnond**
# Description

## Checklist for all submissions

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you tested and reviewed your changes with confidence that everything works?
* [X] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [X] Does your submission pass tests (server is connectable)?
2. [X] Does your egg use a custom docker image?
    * [X] Have you tried to use a generic image?
    * [X] Did you PR the necessary changes to make it work?
3. [X] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [X] Have you added a unique README.md for the egg you are adding according to the alphabetical order?

### CODE REVIEW

**Comment by Rayrnond** [comment]
This PR adds support for GitHub Self-Runner Actions.

**Comment by QuintenQVD0** [comment]
This is sort of like a temp worker LXC container? just like you get with github actions?

**Comment by QuintenQVD0** [comment]
Also a egg should never be locked yo 1 version of the software it can run

**Comment by QuintenQVD0** [comment]
Install images as buser? use the debian 11 one
also running a script as startup is a no go

**Comment by Rayrnond** [comment]
> Install images as buser? use the debian 11 one also running a script as startup is a no go

I'll look into your comments, thank you!

Also the script needs to run ./config which can only be ran once, so I have it ran on installation. If you have any ideas as to a different solution, I'm all ears!

**Comment by QuintenQVD0** [comment]
I made a PR to sort of clean up the install script, have no more time atm will have a look later at this

File: software/github-actions/egg-github-actions.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    "exported_at": "2022-12-24T15:03:37+01:00",
+    "name": "GitHub Actions",
+    "author": "raymond@reflexian.com",
+    "description": "Allows your to host your GitHub Self-Hosted Action runners on Pterodactyl",
+    "features": null,
+    "docker_images": {
+        "ghcr.io\/parkervcp\/yolks:debian": "ghcr.io\/parkervcp\/yolks:debian"
+    },
+    "file_denylist": [],
+    "startup": ".\/run.sh",
```
Comment (parkervcp) [inline software/github-actions/egg-github-actions.json:16]: I am not a huge fan of shell scripts as the startup for a server.

Comment (Xen0Xys) [reply software/github-actions/egg-github-actions.json:16]: The problem is that github runners are started using a script. 
So do you have a prefered way for this case ?

Comment (QuintenQVD0) [reply software/github-actions/egg-github-actions.json:16]: Try to do what they do in that script in the startup cmd. Else you will have to pr it in to the needed image

**Comment by QuintenQVD0** [comment]
This needs Docker to run and docker in docker is a no go.

**Pull request closed without merging**